pkg/provider/provider.go

```diff
@@ -533,7 +539,8 @@ func (k *kubeProvider) Read(ctx context.Context, req *pulumirpc.ReadRequest) (*p
 		// initialize.
 	}
 
-	id := client.FqObjName(liveObj)
+	// TODO(lblackstone): not sure why this is needed
```

Comment:
Me neither.

Reply:
I figured I'd go back over the minor TODOs in a separate PR.